upstream(node): Porting node team related upstream changes in range mainnet-1.34.2 to mainnet-1.35.4

muXxer opened the pull request:
# Description of change

This PR contains a batch of node-team related upstream changes. The changes have been individually reviewed before squash-merging. None of them modify protocol parameters, so a new protocol version is not necessary.

This PR should be rebased on top of develop without squashing as it already contains squashed PRs.

## Links to any relevant issues

List of included PRs:

- #6872
- #6896
- #6897
- #6905
- #6904
- #6912
- #6928
- #6937
- #6907

List of included upstream commits:

- https://github.com/MystenLabs/sui/commit/17aebf1e01c108e21f977f984481bb9b2c6e514e
- https://github.com/MystenLabs/sui/commit/aa71f238844d67ac9fc40849443e370b83306fad
- https://github.com/MystenLabs/sui/commit/16e6c7b69d712712085013afab128d38d09e8d1b
- https://github.com/MystenLabs/sui/commit/3ced52d87a6748ebaf7cdf79b1a679ccfa23d472
- https://github.com/MystenLabs/sui/commit/c76e196213c550b6cfec2b7d3d635cd70cc8a454
- https://github.com/MystenLabs/sui/commit/68497e32faf369ede9bbfb7ab7487c46058ff3d2
- https://github.com/MystenLabs/sui/commit/96b5a149dff31731453baf59fe67e33ccb9fdd27
- https://github.com/MystenLabs/sui/commit/a942c6e243fe5267474d9ecce5bc703b9644bc19
- https://github.com/MystenLabs/sui/commit/8761bf8b70de387bf9ec35025493688482793c60
- https://github.com/MystenLabs/sui/commit/8ecb26f2e7115930b9931f382b87367bca1514b5

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

### Release Notes

- [ ] Protocol:
- [x] Nodes (Validators and Full nodes):
  - Limit the number of peers shared during discovery
  - `TrafficController`: Support allowlisting
  - Add `CheckpointData` to BCS description file
  - Support signed mode in Discovery protocol
  - Enable `WritebackCache` by default
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

gokhan-simsek-iota committed:
node-upstream-change([v1.34.2, v1.35.4)): [iota-node] Add heath check router to JSON-RPC server (#6872)

## Description of change

Port
[MystenLabs/sui@17aebf1](https://github.com/MystenLabs/sui/commit/17aebf1e01c108e21f977f984481bb9b2c6e514e)
Add heath check router to jsonrpc server

## Links to any relevant issues

Part of https://github.com/iotaledger/iota/issues/3990

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

```
cargo r --bin iota-node -- --config-path fullnode.yaml
```

```
$ curl localhost:9000/health 
up

$ curl 'localhost:9000/health?threshold_seconds=10000'
down

$ curl 'localhost:9000/health?threshold_seconds=1000000000'
up
```

### Release Notes

- [x] JSON-RPC: feature: adds a configurable health check endpoint for
json rpc fullnodes to report down if too far behind

bingyanglin committed:
node-upstream-change([v1.34.2, v1.35.4)): Add an option to disable fullnode pruning in test cluster (#6896)

# Description of change

- Port Sui's commit [Add an option to disable fullnode pruning in test
cluster
(#19490)](https://github.com/MystenLabs/sui/commit/aa71f238844d67ac9fc40849443e370b83306fad)
- Currently we aggressive prine FNs by default which causes tests which
depend on reading checkpoints from rest api fail

## Links to any relevant issues

Part of #3990 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran the local network with `RUST_LOG=info cargo run --release --bin iota
start --force-regenesis --with-faucet`

bingyanglin committed:
node-upstream-change([v1.34.2, v1.35.4)): discovery: limit the number of peers shared during discovery (#6897)

# Description of change

- Port Sui's commit [discovery: limit the number of peers shared during
discovery
(#19501)](https://github.com/MystenLabs/sui/commit/16e6c7b69d712712085013afab128d38d09e8d1b)

## Links to any relevant issues

Part of #3990 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran the local network with `RUST_LOG=info cargo run --release --bin iota
start --force-regenesis --with-faucet`

bingyanglin committed:
fix(metrics): Improve metrics push (#6905)

# Description of change

- Upstream range: [v1.34.2, v1.35.4)
- Port Sui's commit [Improve metrics push
(#19460)](https://github.com/MystenLabs/sui/commit/3ced52d87a6748ebaf7cdf79b1a679ccfa23d472)
- A few improvements to metrics push logic.
- Note: Part of the retry logics was merged in our develop branch
already, so our `iota-node/src/metrics.rs` remains untouched.

## Links to any relevant issues

Part of #3990 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran the local network with `RUST_LOG=info cargo run --release --bin iota
start --force-regenesis --with-faucet`

nmrshll committed:
feat(core): [TrafficControl] Support allowlisting (#6904)

## Port
[MystenLabs/sui@c76e196](https://github.com/MystenLabs/sui/commit/c76e196213c550b6cfec2b7d3d635cd70cc8a454)
: [TrafficControl] Support allowlisting
Upstream range: [1.34.2, 1.35.4) 
### Upstream PR(s): 
- https://github.com/MystenLabs/sui/pull/19242  


## Description 

In some cases, we may want to enable a more restrictive policy wherein
the node must explicitly specify all IP's from which it will accept
requests. Because `TrafficController` policies do not easily support
allowlisting, this is instead supported by introducing a separate mode
of operation for traffic controller, enabled by providing `allow_list`
field in the `PolicyConfig`, which should map to a list of strings all
parseable to `IpAddr`.

When this config is present, we skip spawning a tally thread and ignore
all calls to `TrafficController::tally`, and instead initialize an in
memory allowlist of IPs. On subsequent calls to
`TrafficController::check`, we check this list against the requestor IP.

Dry run mode in this mode still works as expected, as do block metrics
(any request that is not in the allowlist is tallied against the block
metric).

gokhan-simsek-iota committed:
upstream(node): add CheckpointData to bcs iota.yaml description file (#6912)

## Description of change

Upstream range: [v1.34.2, v1.35.4)

Port
[MystenLabs/sui@68497e3](https://github.com/MystenLabs/sui/commit/68497e32faf369ede9bbfb7ab7487c46058ff3d2)
Add CheckpointData to bcs iota.yaml description file

## Links to any relevant issues

Part of https://github.com/iotaledger/iota/issues/3990

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

```
cargo r --bin iota-node -- --config-path fullnode.yaml
```

---------

Co-authored-by: jkrvivian <jkrvivian@gmail.com>

nmrshll committed:
upstream(node): Add mode to support signed variant of Discovery protocol (#6928)

## Port
[MystenLabs/sui@96b5a14](https://github.com/MystenLabs/sui/commit/96b5a149dff31731453baf59fe67e33ccb9fdd27):
Add mode to support signed variant of Discovery protocol.
Upstream range: [1.34.2, 1.35.4) 
### Upstream PR(s): 
- https://github.com/MystenLabs/sui/pull/19587  

## Description 

Adds a new version of discovery where all NodeInfo messages are signed
by the originator.

nmrshll committed:
upstream(node): Update outdated event id doc (#6937)

## Port
[MystenLabs/sui@a942c6e](https://github.com/MystenLabs/sui/commit/a942c6e243fe5267474d9ecce5bc703b9644bc19):
Update outdated event id doc
Upstream range: [1.34.2, 1.35.4) 
### Upstream PR(s): 
- https://github.com/MystenLabs/sui/pull/19551  

---------

Co-authored-by: Bing-Yang Lin <bingyang.lin@iota.org>

Thoralf-M reviewed (approved):
Approving for the files I own

bingyanglin committed:
upstream(node): Enable writeback cache by default (#6907)

# Description of change

- Upstream range: [v1.34.2, v1.35.4)
- Port Sui's commit [Enable writeback cache by default
(#19508)](https://github.com/MystenLabs/sui/commit/8761bf8b70de387bf9ec35025493688482793c60)
and [Remove debug output
(https://github.com/MystenLabs/sui/pull/19730)](https://github.com/MystenLabs/sui/commit/8ecb26f2e7115930b9931f382b87367bca1514b5)
- Writeback cache has been stable on all mysten fullnodes plus our
testnet validator for several months. Time to enable it by default
everywhere
- Note that `iter_live_object_set` and
`accumulate_cached_live_object_set_for_testing` do not include
`wrapped_objects` anymore, which was addressed in
https://github.com/iotaledger/iota/pull/3366

## Links to any relevant issues

Part of #3990 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran the local network with `RUST_LOG=info cargo run --release --bin iota
start --force-regenesis --with-faucet`

---------

Co-authored-by: jkrvivian <jkrvivian@gmail.com>
Co-authored-by: Vlad Semenov <vlad.semenov@iota.org>
Co-authored-by: muXxer <git@muxxer.de>

kodemartin reviewed (approved):
lgtm
